src/components/experiment-tracking/details/index.js

```diff
@@ -25,20 +35,50 @@ const Details = ({ selectedRuns, sidebarVisible }) => {
       variables: { runIds: selectedRuns, showDiff: false },
     });
 
+  useEffect(() => {
+    if (!runMetadata) {
+      return;
+    }
+
+    if (!enableComparisonView) {
+      const metadata = runMetadata.find((run) => run.id === selectedRuns[0]);
+
+      setSelectedRunMetadata(metadata);
+    } else {
+      const metadata = runMetadata.find(
+        (run) => run.id === selectedMetadataRunId
+      );
+
+      setSelectedRunMetadata(metadata);
+    }
+  }, [enableComparisonView, runMetadata, selectedMetadataRunId, selectedRuns]);
```

Comment:
we would likely not need this if we omit the `selectedMetadataRunId` hook and assign the run directly within the click callback

Reply:
Brilliant suggestion here as well. I've implemented.